src/core/Song.cpp

```diff
@@ -1026,15 +1026,23 @@ void Song::loadProject( const QString & _file_name )
 
 	emit projectLoaded();
 
-	if( Engine::mainWindow() )
+	if ( hasErrors())
 	{
-		Engine::mainWindow()->showErrors( tr( "The following errors occured while loading: " ) );
+		if ( Engine::hasGUI() )
+		{
+			QMessageBox::warning( NULL, "LMMS Error report", *errorSummary(),
+							QMessageBox::Ok );
+		}
+		else
+		{
+			QTextStream(stderr) << *Engine::getSong()->errorSummary() << endl;
```

Comment:
@diizy This is the best approach I found so far and it's working fine.

It does not use the `<QDebug>` functions and looks more elegant than using `fputf`.